### PR TITLE
Update watch history only when use history

### PR DESCRIPTION
**Pull request by regseb**
With this modification the video is not added to the YouTube history when the video is played (by a connected user) in incognito or _Use playback history_ is disabled.

### CODE REVIEW

**Comment by anxdpanic** [comment]
Thank you for the contribution @regseb  👍 

```use_playback_history()``` is a setting that will actually be removed, it was added to resolve an issue that is no longer present.

Can you change: https://github.com/jdf76/plugin.video.youtube/blob/master/resources/lib/youtube_plugin/youtube/helper/yt_play.py#L78
to
```
use_history = not is_live and not screensaver and not incognito
playback_history = use_history and settings.use_playback_history()
```
and add ```playback_history``` to https://github.com/jdf76/plugin.video.youtube/blob/master/resources/lib/youtube_plugin/youtube/helper/yt_play.py#L102
and
https://github.com/jdf76/plugin.video.youtube/blob/master/resources/lib/youtube_plugin/kodion/utils/player.py#L53

then change 
https://github.com/jdf76/plugin.video.youtube/blob/master/resources/lib/youtube_plugin/kodion/utils/player.py#L53
to ```playback_history```

**Comment by regseb** [comment]
@anxdpanic I added your feedback.

File: resources/lib/youtube_plugin/kodion/utils/player.py [hunk 1]
```diff
@@ -98,7 +99,7 @@ def run(self):
         client = self.provider.get_client(self.context)
         is_logged_in = self.provider.is_logged_in()
 
-        if is_logged_in and report_url:
+        if is_logged_in and report_url and playback_history:
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/kodion/utils/player.py]: Sorry for the minor miscommunication, this should be ```use_history``` still

File: resources/lib/youtube_plugin/kodion/utils/player.py [hunk 2]
```diff
@@ -181,7 +182,7 @@ def run(self):
                 self.update_times(last_total_time, last_current_time, last_segment_start, last_percent_complete)
                 break
 
-            if is_logged_in and report_url:
+            if is_logged_in and report_url and playback_history:
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/kodion/utils/player.py]: this should be ```use_history``` still

File: resources/lib/youtube_plugin/kodion/utils/player.py [hunk 3]
```diff
@@ -226,7 +227,7 @@ def run(self):
 
             p_waited += p_wait_time
 
-        if is_logged_in and report_url:
+        if is_logged_in and report_url and playback_history:
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/kodion/utils/player.py]: this should be ```use_history``` still

File: resources/lib/youtube_plugin/kodion/utils/player.py [hunk 4]
```diff
@@ -250,15 +251,15 @@ def run(self):
         if self.percent_complete >= settings.get_play_count_min_percent():
             play_count = '1'
             self.current_time = 0.0
-            if is_logged_in and report_url:
+            if is_logged_in and report_url and playback_history:
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/kodion/utils/player.py]: this should be ```use_history``` still

File: resources/lib/youtube_plugin/kodion/utils/player.py [hunk 5]
```diff
@@ -250,15 +251,15 @@ def run(self):
...
                 client.update_watch_history(self.video_id, report_url
                                             .format(st=format(self.total_time, '.3f'),
                                                     et=format(self.total_time, '.3f'),
                                                     state=state))
                 self.context.log_debug('Playback reported [%s] @ 100%% state=%s' % (self.video_id, state))
 
         else:
-            if is_logged_in and report_url:
+            if is_logged_in and report_url and playback_history:
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/kodion/utils/player.py]: this should be ```use_history``` still

**Comment by regseb** [comment]
No problem. I corrected `use_history` / `playback_history`.

**Comment by anxdpanic** [comment]
Thanks again :)

**Pull request merged**